File: Lessons/Lesson - 1 - Iterations/Binary_Gap.js
function solution(N) {

    let bin_num = Number(N).toString(2);
    let arr = bin_num.split("");
    
    let bin_group = false;
    let high_zero = 0;
    let zero_counter = 0;

    for (let i of arr) {

        if (i === "1") {
            
            if (high_zero < zero_counter) {
                high_zero = zero_counter;
            };

            bin_group = true;
            zero_counter = 0;

        } else if (bin_group) {
             zero_counter += 1;
        };
    };
    return high_zero;
};

File: Lessons/Lesson - 1 - Iterations/Binary_Gap.py
def solution(N):
    
    bin1 = str(   bin(N)   )  [2:]       # to remove “0b” part
    bin_group = False
    high_zero = 0
    zero_counter = 0
    
    for i in bin1:
        
        if i == "1":
            
            if high_zero < zero_counter:
                high_zero = zero_counter
                
            bin_group = True
            zero_counter = 0
            
        elif bin_group:
            zero_counter += 1
            
    return high_zero
